src/DoubleLinkedList.sol

```diff
@@ -99,47 +96,19 @@ library DoubleLinkedList {
         if (id == address(0)) revert AddressIsZero();
         if (list.accounts[id].value != 0) revert AccountAlreadyInserted();
 
-        uint256 numberOfIterations;
-        address next = list.head; // If not added at the end of the list `id` will be inserted before `next`.
+        address next = getHead(list); // `id` will be inserted before `next`.
 
-        while (numberOfIterations < maxIterations && next != address(0) && list.accounts[next].value >= value) {
-            next = list.accounts[next].next;
-            unchecked {
-                ++numberOfIterations;
-            }
+        uint256 numberOfIterations;
+        for (; numberOfIterations < maxIterations; numberOfIterations++) {
+            if (next == address(0) || list.accounts[next].value < value) break;
+            next = getNext(list, next);
         }
 
-        // Account is not the new tail.
-        if (numberOfIterations < maxIterations && next != address(0)) {
-            // Account is the new head.
-            if (next == list.head) {
-                list.accounts[id] = Account({prev: address(0), next: next, value: value});
-                list.head = id;
-                list.accounts[next].prev = id;
-            }
-            // Account is not the new head.
-            else {
-                address prev = list.accounts[next].prev;
-                list.accounts[id] = Account({prev: prev, next: next, value: value});
-                list.accounts[prev].next = id;
-                list.accounts[next].prev = id;
-            }
-        }
-        // Account is the new tail.
-        else {
-            // Account is the new head.
-            if (list.head == address(0)) {
-                list.accounts[id] = Account({prev: address(0), next: address(0), value: value});
-                list.head = id;
-                list.tail = id;
-            }
-            // Account is not the new head.
-            else {
-                address tail = list.tail;
-                list.accounts[id] = Account({prev: tail, next: address(0), value: value});
-                list.accounts[tail].next = id;
-                list.tail = id;
-            }
-        }
+        if (numberOfIterations == maxIterations) next = address(0);
```

Comment:
nb: it relies on the fact that "numberOfIterations++" is always executed after a loop, wether or not the condition for executing the loop is true or not.

When you think about it, it's actually logical (because the condition can depend on the execution of this expression), and it's the way it works in most langages. But it's not that obvious visually.

an other way to do the same thing, that removes this little thing:
```solidity
while (true) {
    if (next == address(0) || list.accounts[next].value < value) break;
    next = getNext(list, next);    
    numberOfIterations++;
    if (numberOfIterations == maxIterations) {
        next = address(0);
        break;
    }
}
```
also I think that it saves a bit of gas (reusing the end of loop condition to avoid the if after).

I hear that the readability is also somewhat impacted negatively

Reply:
actually it doesn't work if `maxIteration` == 0

Reply:
also, an [decent] optimizing compiler should handle this kind of optimizations.

Reply:
Or we could put the `unchecked { ++numberOfIterations; }` back at the end of the loop. What do you think ?

Reply:
Yes, you need to check first if you reached mach iterations.
So it would give
```solidity
while (true) { 
  if (numberOfIterations == maxIterations) { next = address(0); break; }
  if (next == address(0) || list.accounts[next].value < value) break; 
  next = getNext(list, next); 
  numberOfIterations++;  
}
```
which starts to be very close to the code of the PR.
Maybe an intermediate solution would be 
```solidity
        for (; numberOfIterations < maxIterations;) {
            if (next == address(0) || list.accounts[next].value < value) break;
            next = getNext(list, next);
            numberOfIterations++
        }

        if (numberOfIterations == maxIterations) next = address(0);
 ```
to make it obvious that `numberOfIterations` is incremented. We could even make it `unchecked`, which would be closer to the code before this PR.

What do you think ?

Reply:
>So it would give

Indeed, that works well (and saves one if). I have mixed feelings, I let you choose

Reply:
I'll keep it as is for now. note that we could even do:
```solidity
while (true) { 
  if (numberOfIterations == maxIterations) next = address(0);
  if (next == address(0) || list.accounts[next].value < value) break; 
  next = getNext(list, next); 
  numberOfIterations++;  
}
```